File: app/src/main/java/com/persona5dex/models/room/SearchSuggestion.java
package com.persona5dex.models.room;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Rechee on 11/27/2017.
 */

@Entity(
        tableName = "searchSuggestions"

)
public class SearchSuggestion {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    public int id;

    @ColumnInfo(name = "suggest_text_1")
    public String suggestionColumnTextOne;

    @ColumnInfo(name = "suggest_text_2")
    public String suggestionColumnTextTwo;

    @ColumnInfo(name = "suggest_intent_data")
    public String suggestColumnIntentData;

    @ColumnInfo(name = "suggest_intent_extra_data")
    public String suggestColumnIntentExtraData;
}

File: app/src/main/java/com/persona5dex/searchSuggestions/SearchSuggestion.kt
package com.persona5dex.searchSuggestions

import androidx.room.Entity
import com.persona5dex.models.GameType
import com.persona5dex.models.GameTypePersona

@Entity
open class SearchSuggestion(
        val id: Int,
        val lineOne: String,
        val lineTwo: String,
        val type: Int
)

@Entity
class PersonaSearchSuggestion(
        id: Int,
        lineOne: String,
        lineTwo: String,
        type: Int, override val gameId: GameType

) : SearchSuggestion(id, lineOne, lineTwo, type), GameTypePersona {
    override val name: String
        get() = lineOne
}